### PR TITLE
Add pdu support

### PR DESCRIPTION
**Pull request by cxhong**
For issue #62 Add PDU support.

currently xCAT supports following commands in this release:
compute node based, the pdu attribute has to be defined in the node definition: 
````
rpower cn1 pduoff
rpower cn1 pduon
rpower cn1 pdustat
````
pdu based:
```` 
rpower pdu1 off   ---> turn off all outlets in the pdu
rpower pdu1 on    ---> turn on all outlets in the pdu
rpower pdu1 stat  ---> status for all outlets in the pdu
rinv pdu  ---> status for all outlets in the pdu
````

### CODE REVIEW

**Comment by cxhong** [comment]
Do not merge yet,  I need to add pdu documentation

File: xCAT-server/lib/xcat/plugins/esx.pm [hunk 1]
```diff
@@ -406,6 +406,10 @@ sub process_request {
         @exargs = ($request->{arg});
     }
 
+    #pdu commands will be handled in the pdu plugin
+    if ($command eq "rpower" and grep(/^pduon|pduoff|pdustat/, @exargs)) {
```
Comment (hu-weihua) [inline xCAT-server/lib/xcat/plugins/esx.pm]: Should we update this line to below format like other files?  add ``$`` in the end?

     if ($command eq "rpower" and grep(/^pduon|pduoff|pdustat$/, @exargs)) {

Comment (cxhong) [inline xCAT-server/lib/xcat/plugins/esx.pm]: yes

**Comment by whowutwut** [comment]
I have reviewed the documentation commit of this pull request and I am good with the doc changes.

**Comment by hu-weihua** [comment]
Hi @cxhong,  thanks for your coding. I have reviewed the code, just one problem in line 3665 of ``xCAT-server/lib/xcat/plugins/kvm.pm`` file. I think it's better for us to add ``$`` in expression.

    if ($command eq "rpower" and grep(/^pduon|pduoff|pdustat$/, @exargs)) {
     
But I thought it is not a serious problem. it won't effect the main function. So I decide to merge the code first and let FVT involve ASAP. We can change the ``$`` problem during one bug fixing or pull a new request to fix it if there is no bug opened.   In this way, we can move this work forward quicker.

**Pull request merged**